tests/evm/test_begin_tx.py

```diff
@@ -42,7 +46,7 @@ def test_begin_tx(tx: Transaction, result: bool):
     block = Block()
     caller_balance_prev = rlc_store.to_rlc(int(1e20), 32)
     callee_balance_prev = rlc_store.to_rlc(0, 32)
-    caller_balance = rlc_store.to_rlc(int(1e20) - (tx.value + tx.gas * tx.gas_fee_cap), 32)
+    caller_balance = rlc_store.to_rlc(int(1e20) - (tx.value + tx.gas * tx.gas_price), 32)
     callee_balance = rlc_store.to_rlc(tx.value, 32)
 
     bytecode = Bytecode("00")
```

Comment:
I understand for begin_tx ,the byte code `bytecode = Bytecode("00")` here doesn't matter for any input , right ?

Reply:
Here is a decision I haven't figured it out yet, thanks for reminding me, I will convert this into a issue ASAP.

The decision is, if `callee.code_hash = keccak256('')` (e.g. `callee` is not yet a contract or precompiled), and when there is call to it, we have 2 options:

1. Only do ether transfer, and not dive into another call.
2. Do ether transfer, and prepare the call context just like we are diving into another call. The first executed opcode will always be `STOP` because `program_counter == code_length` (which is not yet implemented).

The first option is the exact behavior of `geth` ([ref](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L219-L220)), but it introduces a branch if `callee.code == keccak256('')`, which requires some more constraints.

The second option is somehow weird, but it doesn't need extra constraint of `BeginTx` or `CALL` in the first option, by sacrificing an extra step, which I think is okay to do.

So the testing now have `Bytecode("00")`, which is actually having the exact behavior of the second option, to temporarily avoid this decision.

Reply:
yeah, we can think more later, I can approve this pr first for now.